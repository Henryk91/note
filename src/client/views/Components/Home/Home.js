import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export default class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            notes: null,
            list: null
        }
    }
    render() {
        let list = this.props.notes
        return (
            <div id="home">
                {list ?
                    <div><br />{createList(list)}</div> :
                    <h3>Please add your name at the top then <br /> click My Notes or All Notes.</h3>
                }
            </div>
        );
    }
}
const createList = (notes) => {
    let list = null
    if (notes) {
        list = notes.map((person) => {
            return (
                <div className="listNameButton" key={person.id}>
                    <Link style={{ textDecoration: 'none' }} to={`/notes/${person.id}`}>
                        <h3>{person.heading}</h3>
                    </Link>
                </div>)
        })
    }
    return list
}
