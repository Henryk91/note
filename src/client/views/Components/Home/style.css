
.listNameButton{
  background: rgb(241, 241, 241);
  border: 1px solid rgb(185, 185, 185);
  margin-top: 2px;
  width: 80vw;
  margin-left: 10vw;
}

@media(min-width: 1000px){
  .listNameButton{
    width: 40vw;
    margin-left: 30vw;
  }
}