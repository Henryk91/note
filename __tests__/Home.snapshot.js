
import Home from '../src/client/views/Components/Home/Home';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { noteTestData } from "../NoteTestData"

test('NoteDetail component renders all details', () => {
    const cData = noteTestData();
    const rendered = renderer.create(
        <Router>
            <Home notes={cData} />
        </Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});