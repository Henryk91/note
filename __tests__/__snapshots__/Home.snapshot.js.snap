// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDetail component renders all details 1`] = `
<div
  id="home"
>
  <div>
    <br />
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9iHNbXqwLsYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Bob
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9iHebXqwLaYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Bobb
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9iHNwXqwLaYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Boby
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9iHNbXqxLaYBz2QUM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Bobyy
           
          Hope
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9qHNbXqwLaYBz2QUq"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Stevee
           
          Jobs
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9iHNbXqwgaYBz2QUq"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Steve
           
          Jobs
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9gHNbXqwLaYBz2QUq"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Stteve
           
          Jobs
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9iHNbXqwLaYBg2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Walt
           
          Sis
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9igNqXqwLaYBz2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Waltt
           
          Sis
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9iHNbgqwLaYBz2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Walter
           
          Sis
        </h3>
      </a>
    </div>
    <div
      className="listNameButton"
    >
      <a
        href="/notes/wCQ9igNbXqwLaYBz2sdM"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <h3>
          Walty
           
          Sis
        </h3>
      </a>
    </div>
  </div>
</div>
`;
